### ISSUE TITLE
SafetyNet / Play Integrity column

### ISSUE THREAD

**Issue by D3SOX** [eylenburg/eylenburg.github.io#33]
I was wondering why it says Yes for CalyxOS as I'm running the latest build on my Pixel 8 Pro and I get neither SafetyNet nor Play Integrity. As far as I know it's not possible to pass it on a custom ROM without using something like Magisk and a module that fixes it (not sure about Graphene)

**Comment by eylenburg** [comment]
I think you're right. 

For GrapheneOS [[1]](https://grapheneos.org/usage#banking-apps) it says that only `basicIntegrity` is passed although it doesn't mention Play Integrity here but only Safetynet.

For microG I believe it's the same (only `basicIntegrity`) and the additional issue that Play Integrity depends on the Play Store? [[2]](https://github.com/microg/GmsCore/wiki/Implementation-Status) [[3]](https://github.com/microg/GmsCore/issues/2168) [[4]](https://github.com/microg/GmsCore/issues/1875) [[5]](https://github.com/microg/GmsCore/issues/2050) [[6]](https://github.com/microg/GmsCore/pull/2077)

What would you propose is the right way to show it?


  | GrapheneOS Play Services | MicroG | MicroG + root
-- | -- | -- | --
SafetyNet basic integrity | OK | OK | OK
CTS Profile match | No | No | OK ???
Play Integrity | No | No | OK ???

Is that correct?

**Comment by eylenburg** [comment]
Thank you @matchboxbananasynergy 

Do you know if that's the same for microG?

**Comment by eylenburg** [comment]
I agree. I just updated the row to saw "passes only basic integrity" in light green. The only one exception being Stock Android of course.

**Comment by ale5000-git** [comment]
I haven't tried it but I don't think it is impossible to pass strong integrity (but only if it is done directly by ROM authors).

For ROMs that spoof original details (like model, device, fingerprint, etc.) but also Kernel version strings, and they are even able to relock the bootloader with the cutom ROM; then maybe they can pass without root and without Magisk.

**Comment by ale5000-git** [comment]
> It's not possible is hardware attestation is used. Play integrity API is moving to that, and it won't be spoofable, not matter what the OS does.

If I'm not wrong hardware attestation works without any problem if the ROM relock the bootloader and don't include root or Magisk.

**Comment by ale5000-git** [comment]
Doesn't Google Pixel devices allow to relock the bootloader without the stock OS?

**Comment by ale5000-git** [comment]
> It checks for the stock OS with hardware attestation rather than an aftermarket OS with a specific verified boot key.

If the OS itself spoof data of a stock OS and replace yellow with green at boot (from apps view without Magisk or root) and lock the boot-loader wouldn't it pass?

**Comment by ale5000-git** [comment]
I'm not too much expert in this field but nothing is really impossible; it is just that many times isn't worth the efforts.

For example another idea is: having the untouched stock OS in the system partition ans than with an exploit during boot unload the OS and load another one in another partition under /data so that /system in untouched.
Just a rough idea, I haven't checked the feasibility and it is unlikely that I will look at it; it is just to clear the point that with enough fantasy and free time anithing can be done.

**Comment by ale5000-git** [comment]
As I said it was just an example that if you cannot enter in one place from a door you can use the window, the final point is just to pass Play Integrity from the app view.
If it is real or fake it doesn't matter.
